app/controllers/posts_controller.rb

```diff
@@ -23,9 +23,10 @@ def create
     respond_to do |format|
       format.html do
         if new_post.save
-          redirect_to "/users/#{new_post.user.id}/posts/", notice: 'Success!'
+          redirect_to "/users/#{new_post.user.id}/posts/", flash: { notice: 'Success! Post has been created!' }
         else
-          render :new, alert: 'Error!'
+          render :new, flash.now[:error] = "Error, post couldn't be saved!"
+          render action: 'new'
         end
       end
```

Comment:
I get the same error as pointed out by the previous code reviewer but it occurs when a user clicks on a post after he tries to add a comment on a post.
![post-show](https://user-images.githubusercontent.com/35504149/156799564-6f4032ba-6ae3-4f1d-b527-233cc84e6650.png)